### DISCUSSION TITLE
v5.0.x development

### DISCUSSION THREAD

**Discussion by swurzinger** [Blair2004/NexoPOS#1338] [General]
Hi,

I'd like to continue my development activities on the v5.0.x branch if possible, but saw that that branch hasn't been updated for a while. e.g. all the library upgrades that were automatically done by dependabot have been merged into master, but not into the v5 branch. Similarily some features/bugs have been added/fixed on master or a v4.x branch, but not merged into v5.

Can we merge current master into v5?
I already did that in https://github.com/swurzinger/NexoPOS/tree/stw/5 resolving all the merge conflicts, but it still has some issues that need to be fixed and probably were merged incorrectly (by me). Before I continue working on that I have some questions about the future development plans:

@Blair2004 
What are the plans regarding the next version? Is there any release schedule? Is there any list of TODOs / features / requirements / tickets that still need to be implemented till v5 will be ready?

Do you regularly integrate master into v5? Does it make sense if I continue on the merge or will you do it anyways in the near future? And if I continued, would you accept a PR for the merge?

Thanks!

**Comment by Blair2004** [comment]
Hey,
That's great you stepped on v5. First of all that branch is on sleep mainly because I'm currently creating a ReactJs - ELECTRON app for Cloud Printing.

I am integrating that app with my.nexopos.con and it takes some time as I'm new on ReactJs.

So, we can't merge v5 on master as it's not stable right now. There is indeed a lot of work, but slowing those are getting done. At the moment I think most of the desired features are added to v5. I think we only need to clean using your recent PR (I can handle that with cherry-pick)

I can't have a release date, because I'm not sure I can have the project ready at the moment of publishing. So I set a date, but I'm most of the time overpassing that.

Hope I answered your questions

**Comment by Blair2004** [comment]
Hey,
That's great you stepped on v5. First of all that branch is on sleep mainly because I'm currently creating a ReactJs - ELECTRON app for Cloud Printing.

I am integrating that app with my.nexopos.con and it takes some time as I'm new on ReactJs.

So, we can't merge v5 on master as it's not stable right now. There is indeed a lot of work, but slowing those are getting done. At the moment I think most of the desired features are added to v5. I think we only need to clean using your recent PR (I can handle that with cherry-pick)

I can't have a release date, because I'm not sure I can have the project ready at the moment of publishing. So I set a date, but I'm most of the time overpassing that.

It makes more sense if your new PR are on v5 (otherwise I'll be forced to cherry pick that)

Hope I answered your questions.

**Reply by swurzinger** [reply]
Hi,

Thanks for the reply!

I thought about merging master into v5, not the other way round (as v5 is not ready yet as far as I understood). Or do you just want to cherry-pick specific commits from master/v4.x into v5? Unfortunately most of the commit messages are bad and do not describe what has been changed. (Also existing commits tend to be quite large.)

I'll base my changes on v5 then.

**Reply by Blair2004** [reply]
We need to cherry-pick specific commits from master to v5. Regarding the message, I most of the time describe what happened with a commit. Maybe you're talking about other commits. While exiting commit seems large, I believe it's the way to go (cherry-pick).

Thanks
